remoteworkitem/remoteworkitem_test.go

```diff
@@ -75,160 +75,238 @@ func TestWorkItemMapping(t *testing.T) {
+		remoteWorkItemImpl := RemoteWorkItemImplRegistry[ProviderGithub]
+		gh, err := remoteWorkItemImpl(remoteTrackerkItem)
+
+		if err != nil {
+			t.Fatal(err)
+		}
+		workItem, err := Map(gh, workItemMap)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, localWorkItemKey := range workItemMap {
+			t.Log("Mapping ", localWorkItemKey)
+			_, ok := workItem.Fields[localWorkItemKey]
+			assert.Equal(t, ok, j.expectedOutput, fmt.Sprintf("%s not mapped", localWorkItemKey))
+		}
 	}
 }
 func TestJiraIssueMapping(t *testing.T) {
 	resource.Require(t, resource.UnitTest)
 
-	content, err := test.LoadTestData("jira_issue_mapping.json", provideRemoteJiraDataWithAssignee)
-	if err != nil {
-		t.Fatal(err)
+	type jiraData struct {
+		inputFile      string
+		expectedOutput bool
 	}
 
-	workItemMap := WorkItemKeyMaps[ProviderJira]
-	remoteTrackerItem := TrackerItem{Item: string(content[:]), RemoteItemID: "xyz", TrackerID: uint64(0)}
-	remoteWorkItemImpl := RemoteWorkItemImplRegistry[ProviderJira]
-	ji, err := remoteWorkItemImpl(remoteTrackerItem)
-
-	if err != nil {
-		t.Fatal(err)
-	}
-	workItem, err := Map(ji, workItemMap)
-	if err != nil {
-		t.Fatal(err)
+	var jir = []jiraData{
+		{"jira_issue_mapping.json", true},
 	}
 
-	for _, localWorkItemKey := range workItemMap {
-		t.Log("Mapping ", localWorkItemKey)
-		_, ok := workItem.Fields[localWorkItemKey]
-		assert.Equal(t, ok, true, fmt.Sprintf("%s not mapped", localWorkItemKey))
+	for _, j := range jir {
+		content, err := test.LoadTestData(j.inputFile, provideRemoteJiraDataWithAssignee)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		workItemMap := WorkItemKeyMaps[ProviderJira]
+		remoteTrackerItem := TrackerItem{Item: string(content[:]), RemoteItemID: "xyz", TrackerID: uint64(0)}
+		remoteWorkItemImpl := RemoteWorkItemImplRegistry[ProviderJira]
+		ji, err := remoteWorkItemImpl(remoteTrackerItem)
+
+		if err != nil {
+			t.Fatal(err)
+		}
+		workItem, err := Map(ji, workItemMap)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, localWorkItemKey := range workItemMap {
+			t.Log("Mapping ", localWorkItemKey)
+			_, ok := workItem.Fields[localWorkItemKey]
+			assert.Equal(t, ok, j.expectedOutput, fmt.Sprintf("%s not mapped", localWorkItemKey))
+		}
 	}
 }
 
 func TestFlattenGithubResponseMap(t *testing.T) {
 	resource.Require(t, resource.UnitTest)
-	testString, err := test.LoadTestData("github_issue_mapping.json", provideRemoteGithubDataWithAssignee)
-	if err != nil {
-		t.Fatal(err)
+
+	type githubData struct {
+		inputFile      string
+		expectedOutput bool
 	}
-	var nestedMap map[string]interface{}
-	err = json.Unmarshal(testString, &nestedMap)
 
-	if err != nil {
-		t.Error("Incorrect dataset ", testString)
+	var gitData = []githubData{
+		{"github_issue_mapping.json", true},
 	}
 
-	OneLevelMap := Flatten(nestedMap)
+	for _, j := range gitData {
+		testString, err := test.LoadTestData(j.inputFile, provideRemoteGithubDataWithAssignee)
```

Comment:
```
func provideRemoteGithubDataWithAssignee() ([]byte, error) {
	return provideRemoteData(GithubIssueWithAssignee)
}
```
This method always loads data from the same url `GithubIssueWithAssignee`. 

For every file, you put inside

var gitData = []githubData{
      {"github_issue_mapping.json", true},
 }
 there should be a different issue url which would get tested.